Source/ARTDefault.m

```diff
@@ -109,4 +136,54 @@ + (NSString *)bundleBuildNumber {
     return infoDictionary[ARTDefault_bundleBuildNumberKey];
 }
 
++ (NSString *)osName {
+    return
+        #if TARGET_OS_IOS
+            @"iOS"
+        #elif TARGET_OS_TV
+            @"tvOS"
+        #elif TARGET_OS_WATCH
+            @"watchOS"
+        #elif TARGET_OS_OSX
+            @"macOS"
+        #else
+            @"Unknown"
+        #endif
+        ;
+}
+
++ (NSString *)osVersionString {
+    static NSString *versionString;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSOperatingSystemVersion version = [[NSProcessInfo processInfo] operatingSystemVersion];
+        versionString = [NSString stringWithFormat:@"%u.%u.%u",
+                         conformVersionComponent(version.majorVersion),
+                         conformVersionComponent(version.minorVersion),
+                         conformVersionComponent(version.patchVersion)];
+    });
+    return versionString;
+}
+
++ (NSString *)deviceModel {
+    struct utsname systemInfo;
+    uname(&systemInfo);
+    return [NSString stringWithCString:systemInfo.machine encoding:NSUTF8StringEncoding];
+}
+
++ (NSString *)deviceModelToken {
+    static NSString *modelToken;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        modelToken = canonizeStringAsUserAgentToken([self deviceModel]);
+    });
+    return modelToken;
+}
+
++ (NSString *)agent {
+    NSString* agentString = [NSString stringWithFormat:@"%@/%@ %@/%@ hw/%@",
```

Comment:
I'm not sure where the `hw/` prefix has come from? The second part of that (`deviceModelToken`) is definitely not SemVer compatible! I think you meant this:

```suggestion
    return [NSString stringWithFormat:@"%@/%@ %@/%@ %@",
```

(no need for `agentString` local variable, either 😉)

Reply:
https://github.com/ably/ably-cocoa/pull/1119/commits/68b1db88db1ce4b54ec9f2715a5761e87c2aa12a